fxa/tests/test_core.py

```diff
@@ -328,3 +330,27 @@ def test_get_identity_assertion_accepts_service(self):
                                                         service="test-me")
         data = browserid.verify(assertion, audience="http://example.com")
         self.assertEquals(data["status"], "okay")
+
+    def test_totp(self):
+        resp = self.session.totp_create()
+
+        # Double create causes a client error
+        with self.assertRaises(fxa.errors.ClientError):
+            self.session.totp_create()
+
+        # Created but not verified returns false
+        self.assertFalse(self.session.totp_exists())
```

Comment:
It took me a long time to figure this out, but I think you're falling afoul of the behaviour described in the comment here:

  https://github.com/mozilla/fxa-auth-server/blob/master/lib/routes/totp.js#L206

Calling `/totp/exists` before verifying the token, causes us to delete the token on the assumption that the user failed the TOTP setup process.  If I remove this line then the tests seem to pass as expected.

Reply:
(It's not obvious to me whether we really need to behave this way, I'm going to file an fxa-auth-server issue to discuss)

Reply:
https://github.com/mozilla/fxa-auth-server/issues/2541